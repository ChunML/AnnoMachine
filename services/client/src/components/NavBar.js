import React from 'react';

function NavBar(props) {
  return (
    <div className="ui inverted menu">
      <a className="active item" href="/">
        Home
      </a>
      <a className="item">
        Profile
      </a>
      <div className="right menu">
        <a className="item">
          Register
        </a>
        <a className="item">
          Log In
        </a>
        <div className="item">
          <div className="ui icon input">
            <input type="text" placeholder="Search tag..." />
            <i className="search icon"></i>
          </div>
        </div>
      </div>
    </div>
  )
}

export default NavBar;