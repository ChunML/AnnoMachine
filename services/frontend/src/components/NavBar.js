import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function NavBar({ title, isAuthenticated }) {
  return (
    <div className="nav flex-center space-around text-center">
      <div className="nav-header">{title}</div>
      <div className="nav-link">
        <Link to="/">Home</Link>
      </div>
      {isAuthenticated && (
        <div className="nav-link">
          <Link to="/logout">Log Out</Link>
        </div>
      )}
      {!isAuthenticated && (
        <React.Fragment>
          <div className="nav-link">
            <Link to="/register">Register</Link>
          </div>
          <div className="nav-link">
            <Link to="/login">Log In</Link>
          </div>
        </React.Fragment>
      )}
      <div className="nav-link">
        <div className="input-field-dark">
          <input type="text" placeholder="Search tag..." />
        </div>
      </div>
    </div>
  );
}

NavBar.propTypes = {
  title: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
};

export default NavBar;
