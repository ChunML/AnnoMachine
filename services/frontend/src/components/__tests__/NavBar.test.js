import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NavBar from '../NavBar';
import '../../setupTests';

const title = 'Hi, there!';

describe('When not authenticated', () => {
  it('Navbar renders properly', () => {
    const wrapper = shallow(<NavBar title={title} isAuthenticated={false} />);
    const header = wrapper.find('.nav-header');
    expect(header.length).toBe(1);
    expect(header.get(0).props.children).toBe(title);

    const items = wrapper.find('.nav-link');
    expect(items.length).toBe(4);
    expect(items.get(0).props.children.props.children).toBe('Home');
    expect(items.get(1).props.children.props.children).toBe('Register');
    expect(items.get(2).props.children.props.children).toBe('Log In');
  });

  it('Navbar renders a snapshot properly', () => {
    const tree = renderer
      .create(
        <Router location="/">
          <NavBar title={title} isAuthenticated={false} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('When authenticated', () => {
  it('Navbar renders properly', () => {
    const wrapper = shallow(<NavBar title={title} isAuthenticated />);
    const header = wrapper.find('.nav-header');
    expect(header.length).toBe(1);
    expect(header.get(0).props.children).toBe(title);

    const items = wrapper.find('.nav-link');
    expect(items.length).toBe(3);
    expect(items.get(0).props.children.props.children).toBe('Home');
    expect(items.get(1).props.children.props.children).toBe('Log Out');
  });

  it('Navbar renders a snapshot properly', () => {
    const tree = renderer
      .create(
        <Router location="/">
          <NavBar title={title} isAuthenticated />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
