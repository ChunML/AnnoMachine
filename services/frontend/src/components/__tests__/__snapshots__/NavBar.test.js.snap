// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When authenticated Navbar renders a snapshot properly 1`] = `
<div
  className="ui inverted menu"
>
  <div
    className="header item"
  >
    Hi, there!
  </div>
  <div
    className="active item"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </div>
  <div
    className="right menu"
  >
    <div
      className="item"
    >
      <a
        href="/logout"
        onClick={[Function]}
      >
        Log Out
      </a>
    </div>
    <div
      className="item"
    >
      <div
        className="ui icon input"
      >
        <input
          placeholder="Search tag..."
          type="text"
        />
        <i
          className="search icon"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`When not authenticated Navbar renders a snapshot properly 1`] = `
<div
  className="ui inverted menu"
>
  <div
    className="header item"
  >
    Hi, there!
  </div>
  <div
    className="active item"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </div>
  <div
    className="right menu"
  >
    <div
      className="item"
    >
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
    <div
      className="item"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </div>
    <div
      className="item"
    >
      <div
        className="ui icon input"
      >
        <input
          placeholder="Search tag..."
          type="text"
        />
        <i
          className="search icon"
        />
      </div>
    </div>
  </div>
</div>
`;
